### PR TITLE
Feature/passthru annotation select

### PR DESCRIPTION
**Pull request by aaronz8**
Added option to override/set id of annotations and to `select` annotations to make them active.

### CODE REVIEW

File: src/visualization/line.js [hunk 1]
```diff
@@ -201,6 +201,10 @@ define([
       }
     },
 
+    select: function(id) {
```
Comment (drakonen) [inline src/visualization/line.js]: function name Select is too ambiguous, you are selecting an annotation.
This is should probably be a viewOptions callback.

File: src/widgets/line.js [hunk 2]
```diff
@@ -9,10 +9,15 @@ define([
       this.colors = this.getColors();
     },
 
-    view: function() {
-      return new Openhose.Visualization.Line(Openhose._.extend({
-        colors: this.getColors()
+    view: function () {
```
Comment (drakonen) [inline src/widgets/line.js]: If you make this a viewOption, this is not needed.

File: src/widgets/line.js [hunk 3]
```diff
@@ -9,10 +9,15 @@ define([
...
+      this.visualization = new Openhose.Visualization.Line(Openhose._.extend({ 
+        colors: this.getColors() 
       }, this.viewOptions));
+      return this.visualization;
+    },
+
+    select: function(id) {
```
Comment (drakonen) [inline src/widgets/line.js]: If you make this a viewOption, this is not needed.

**Comment by drakonen** [comment]
You should add an example.

You can add this select functionality as a callback passed along with the viewOptions object.
Like so:
https://github.com/openhose/openhose-visualizations-sdk/blob/70d1937b99bcd45a2d4eb8090c9c0f5e1fe791d4/src/visualization/line.js#L24
(unfortunately there is no example of this, if you feel adventurous, add it)

On a widget it would look like this:
```
var line = new Openhose.Widget.Line({
  el: Openhose.$(".widget"), // container DOM element
  data: { },
  viewOptions: {
    height: 150,
    someCallbackFunction: someCallbackFunction () {}
  },
  period: period
});
```

On a vis it would look like this:
```
var line = new Openhose.Visualization.Line({
  data: data,
  el: Openhose.$(".line1"),
  period: period,
  height: 150,
  someCallbackFunction: someCallbackFunction () {}
});
```

File: dist/openhose-visualizations-sdk.bundle.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /*
 // Openhose Visualization SDK Kit (Bottlenose)
 // ----------------------------------
-// v0.9.2
```
Comment (Juice10) [inline dist/openhose-visualizations-sdk.bundle.js]: Please don't commit bundled files, this causes merge conflicts. Bundling will be done on master after pull request is accepted

**Comment by aaronz8** [comment]
I realized that since widget.visualization is the visualization, and visualization.annotator is CustomAnnotate plugin, I can just call this.graph.visualization.annotator.select(annotation._id) from wherever it’s needed.

That makes all the `select` relaying functions obsolete and no more use for events. :)

@Juice10 Should I revert the bundled files?

**Comment by aaronz8** [comment]
@drakonen 
I'm not sure how to proceed with writing an example, because I've pretty much removed everything except the option to override id's in visualization/line.js:

    this.annotator.add(annotation.x, annotation.label, end, series, annotation.click, annotation.id);

and that depends on how this.annotator/CustomAnnotate defines `.add`. (not even in openhose-vis)

**Comment by drakonen** [comment]
@aaronz8: you can make an example using it kind of the way you use it. It
can be sort of documentation.

**Comment by Juice10** [comment]
Yes please revert the bundled files

**Comment by aaronz8** [comment]
@Juice10 done with reverts
@drakonen done with examples and moved custom-annotate from NC to oh-vis-sdk.

**Comment by drakonen** [comment]
:+1:

**Pull request merged**